### PR TITLE
[Localization] Localize hp phases and translate in Korean

### PR DESCRIPTION
**Pull request by EnochG1**
## What are the changes?
Add the option to localize message text in phases.ts, 2 HP restore messages.

## Why am I doing these changes?
All text should be localizable and I thought HP message need to translate.

## What did change?
 - Change the code for localize -> messages of phases.ts to add the option to localize
 - Added localization key in each language folders.
 - Translated part of them to Korean.
 - Note : Localized messages are below
```
  "turnEndHpRestore": "{{pokemonName}}'s HP was restored.", // when pokemon healed by TerrainType.GRASSY (GRASSY_TERRAIN)
  "hpIsFull": "{{pokemonName}}'s\nHP is full!", // when pokemon's HP already full
```

### Screenshots/Videos

English message should not be changed.
(If images are too small, just click image)

| Case | Language | As-is | To-be |
| :-----: | :-----: | :-----: | :---: |
| turnEndHpRestore | En | image | image |
| turnEndHpRestore | Ko | image | image |
| turnEndHpRestore | Fr | image | image |
| hpIsFull | En | image | image |
| hpIsFull | Ko | image | image |
| hpIsFull | Fr | image | image |

## How to test the changes?
Manually.
Change override.ts moveset, add Grassy Terrain to make terrain Grassy.
```typescript
export const MOVESET_OVERRIDE: Array<Moves> = [ Moves.GRASSY_TERRAIN ];
```
Then check when turn end.
hpIsFull message is occur when pokemon's HP is full.

## Checklist
- [X] There is no overlap with another PR?
- [X] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [X] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/fr/battle.ts [hunk 1]
```diff
@@ -53,6 +53,8 @@ export const battle: SimpleTranslationEntries = {
   "escapeVerbSwitch": "le changement",
   "escapeVerbFlee": "la fuite",
   "notDisabled": "La capacité {{moveName}}\nde {{pokemonName}} n’est plus sous entrave !",
+  "turnEndHpRestore": "{{pokemonName}}'s HP was restored.",
+  "hpIsFull": "{{pokemonName}}'s\nHP is full!",
```
Comment (Adri1) [inline src/locales/fr/battle.ts:57]:
```suggestion
  "turnEndHpRestore": "{{pokemonName}} récupère des PV !",
  "hpIsFull": "Les PV de {{pokemonName}}\nsont au maximum !",
```

**Review by Adri1 — COMMENTED** [review]
French translation

**Review by sodaMelon — APPROVED** [review]
there's no problem in Korean script. LGTM

**Comment by EnochG1** [comment]
@Adri1 
Could you check result of test and approve if there's no problem?

| turnEndHpRestore | hpIsFull |
| --- | --- |
| image | image |

**Review by Adri1 — APPROVED** [review]

**Review by Tempo-anon — APPROVED** [review]

**Pull request merged**